server/controllers/message.js

```diff
@@ -0,0 +1,59 @@
+const Message = require("../models/Message");
+const Conversation = require("../models/Conversation");
+
+// @route POST /new
+// @desc create a new message associated with a conversation
+// @access Private
+exports.createNewMessage = async (req, res) => {
+  const userId = req.user.id;
+  const { conversationId, content } = req.body;
+
+  if (conversationId && content) {
+    const conversation = await Conversation.findById({ _id: conversationId });
+
+    if (!conversation) {
+      res.status(400).send("Conversation not found.");
+    } else {
+      const newMessage = await Message.create({
+        conversationId,
+        author: userId,
+        content,
+      });
+
+      if (!newMessage) {
+        res.status(500).send("An error occurred while sending a message.");
+      } else {
+        const willRead = conversation.users.filter((user) => String(user) !== String(userId));
```

Comment:
Question: What do we need this `willRead` variable?

Reply:
As per our conversation, the read status has been moved to the Messages model. Thanks for the suggestion!

Reply:
This is no longer needed, as the read status was moved to the Message model.